CMakeLists.txt

```diff
@@ -792,3 +791,15 @@ foreach(file ${ZIG_STD_FILES})
     get_filename_component(file_dir "${ZIG_STD_DEST}/${file}" DIRECTORY)
     install(FILES "${CMAKE_SOURCE_DIR}/std/${file}" DESTINATION "${file_dir}")
 endforeach()
+
+include(CTest)
+if(BUILD_TESTING AND CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+    add_executable(fuzz_test src/fuzz_test.cpp ${ZIG_SOURCES} ${ZIG_CPP_SOURCES})
+    target_compile_options(fuzz_test PRIVATE "-fsanitize=address,undefined,fuzzer")
+    target_link_libraries(fuzz_test PRIVATE "-fsanitize=address,undefined,fuzzer"
```

Comment:
This is currently failing with the xcode build, presumably because xcode doesn't have libFuzzer enabled:
```
/Applications/Xcode.app/Contents/Developer/usr/bin/g++   -I/Users/travis/build/ziglang/zig/deps/lld/include -I/usr/local/Cellar/llvm/6.0.0/include -I/usr/local/opt/llvm@6/include -I/Users/travis/build/ziglang/zig/deps/SoftFloat-3e/source/include -I/Users/travis/build/ziglang/zig -I/Users/travis/build/ziglang/zig/build -I/Users/travis/build/ziglang/zig/src -I/Users/travis/build/ziglang/zig/deps/SoftFloat-3e-prebuilt -I/Users/travis/build/ziglang/zig/deps/SoftFloat-3e/source/8086  -g -Wall   -fsanitize=address,undefined,fuzzer -o CMakeFiles/fuzz_test.dir/src/fuzz_test.cpp.o -c /Users/travis/build/ziglang/zig/src/fuzz_test.cpp
clang: error: unsupported argument 'fuzzer' to option 'fsanitize='
```
Don't know about xcode 10 but with xcode 9, neither clang++ nor g++ understand `-fsanitize=fuzzer`.

Reply:
Mac aliases `g++` to `clang++` so they are actually the same compiler. Ironically, despite Apple's support for clang, Mac seems to be incapable of running the sanitizers/fuzzer by default. I know all about this because I have a Mac laptop.